src/agent/MessageSender.ts

```diff
@@ -22,6 +29,12 @@ class MessageSender {
 
   public async sendMessage(outboundMessage: OutboundMessage): Promise<void> {
     const outboundPackage = await this.envelopeService.packMessage(outboundMessage);
+    const transport = this.transportService.getTransport(outboundMessage.connection.id);
+    if (transport) {
+      outboundPackage.transport = transport;
+    } else {
+      outboundPackage.transport = new HttpTransport(outboundMessage.endpoint);
```

Comment:
This should look at the scheme of the endpoint instead of creating a hardcoded transport. Ideally the `outboundMessage.endpoint` is not set here yet. As we could then dynamically select the transport to use based on available endpoints of the the connection and available transports.